### ISSUE TITLE
Xournalpp crashed on image paste from clipboard

### ISSUE THREAD

**Issue by Esokrates** [xournalpp/xournalpp#3044]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux (Ubuntu 21.04)
 - Desktop environment: Gnome-Wayland
 - libgtk 3.24.25
 - Version of Xournal++: 1.1.0+dev (Git commit 6af02331)
 - Installation method: self compiled from master on Apr. 5

**Describe the bug**

Clicked the Paste button in order to paste an image from clipboard, when Xournalpp crashed. Restarting an trying to paste the same image again did not reproduce  the crash.

Terminal output when the crash happened:

```
(xournalpp:7250): GdkPixbuf-CRITICAL **: 23:57:03.207: gdk_pixbuf_get_width: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(xournalpp:7250): GdkPixbuf-CRITICAL **: 23:57:03.207: gdk_pixbuf_get_height: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(xournalpp:7250): GdkPixbuf-CRITICAL **: 23:57:03.207: gdk_pixbuf_get_pixels_with_length: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(xournalpp:7250): GdkPixbuf-CRITICAL **: 23:57:03.207: gdk_pixbuf_get_rowstride: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(xournalpp:7250): GdkPixbuf-CRITICAL **: 23:57:03.207: gdk_pixbuf_get_n_channels: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

** (xournalpp:7250): WARNING **: 23:57:03.207: [Crash Handler] Crashed with signal 11

** (xournalpp:7250): WARNING **: 23:57:03.230: [Crash Handler] Wrote crash log to: /home/user/.cache/xournalpp/errorlogs/errorlog.20210416-235703.log

** (xournalpp:7250): WARNING **: 23:57:03.377: Trying to emergency save the current open document…

** (xournalpp:7250): WARNING **: 23:57:03.539: Successfully saved document to "/home/user/.config/xournalpp/emergencysave.xopp"
```
Output of `/home/user/.cache/xournalpp/errorlogs/errorlog.20210416-235703.log`:

```
Date: Fri Apr 16 23:57:03 2021
Error: signal 11
[bt]: (0) ./xournalpp(+0x4487e8) [0x55824196a7e8]
[bt]: (1) /lib/x86_64-linux-gnu/libc.so.6(+0x41040) [0x7f5459f84040]
[bt]: (2) ./xournalpp(_Z25f_pixbuf_to_cairo_surfaceP10_GdkPixbuf+0x175) [0x558241981f90]
[bt]: (3) ./xournalpp(_ZN5Image8setImageEP10_GdkPixbuf+0x20) [0x558241929bae]
[bt]: (4) ./xournalpp(_ZN7Control19clipboardPasteImageEP10_GdkPixbuf+0x52) [0x5582417a3b30]
[bt]: (5) ./xournalpp(_ZN16ClipboardHandler19pasteClipboardImageEP13_GtkClipboardP10_GdkPixbufPS_+0x3e) [0x558241796930]
[bt]: (6) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3c83a4) [0x7f545b1053a4]
[bt]: (7) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3c7a67) [0x7f545b104a67]
[bt]: (8) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3f337b) [0x7f545b13037b]
[bt]: (9) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0x1294) [0x7f545a889724]
[bt]: (10) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_by_name+0x1f9) [0x7f545a889aa9]
[bt]: (11) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e0a0d) [0x7f545b01da0d]
[bt]: (12) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3e8e28) [0x7f545b125e28]
[bt]: (13) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0x1294) [0x7f545a889724]
[bt]: (14) /lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit+0x93) [0x7f545a889893]
[bt]: (15) /lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b00d4) [0x7f545b0ed0d4]
[bt]: (16) /lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_main_do_event+0xa5a) [0x7f545af8f73a]
[bt]: (17) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0x39733) [0x7f545ac71733]
[bt]: (18) /lib/x86_64-linux-gnu/libgdk-3.so.0(+0xa03d6) [0x7f545acd83d6]
[bt]: (19) /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x25b) [0x7f545b54184b]
[bt]: (20) /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0xa8c68) [0x7f545b594c68]
[bt]: (21) /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x33) [0x7f545b53ef83]
[bt]: (22) /lib/x86_64-linux-gnu/libgio-2.0.so.0(g_application_run+0x1cd) [0x7f545a98f27d]
[bt]: (23) ./xournalpp(_ZN11XournalMain3runEiPPc+0x5ef) [0x5582417d1fc6]
[bt]: (24) ./xournalpp(main+0x29) [0x55824178f1b2]
[bt]: (25) /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xd5) [0x7f5459f6b565]
[bt]: (26) ./xournalpp(_start+0x2e) [0x55824178f0ce]
```
I know that backtrace is not symbolic, but is there something I could do? Does xournalpp dump core somwhere so that I could redo a backtrace with all symbols and gdb? Otherwise we have a problem here, since the crash was not reproducible.

**Comment by rolandlo** [comment]
@Esokrates Could it be that you closed the application from which the image was copied before pasting? That may not be supported. Compare #2262

**Comment by Esokrates** [comment]
@rolandlo Can definitely rule that one out, I took a screenshot using gnome-screenshot, selected "Copy to clipboard" and the gnome-screen application was still open. In fact when I started Xournalpp after the crash I could paste the picture without any problems.

**Comment by Esokrates** [comment]
Looking at the trace of #2262 it looks pretty likely that the are indeed caused by the same bug.

**Comment by rolandlo** [comment]
Please reopen if the bug was not fixed via #4489 (in the nightly build to follow).
